### PR TITLE
fix/PRO-982/minimum-swap-amount-check

### PR DESCRIPTION
**Pull request by Janislav**
# Pull Request

Closes: PRO-982

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have updated documentation where appropriate.

## Summary

*Please include a succinct description of the purpose and content of the PR. What problem does it solve, and how? Link issues, discussions, other PRs, and anything else that will help the reviewer.*

### CODE REVIEW

File: state-chain/pallets/cf-swapping/src/lib.rs [hunk 1]
```diff
@@ -1044,13 +1048,14 @@ pub mod pallet {
...
 				amount,
+				net_amount,
 				destination_address.clone(),
 				&swap_origin,
 			) {
 				Self::deposit_event(Event::<T>::SwapScheduled {
 					swap_id,
 					source_asset: from,
-					deposit_amount: amount,
+					deposit_amount: net_amount,
```
Comment (dandanlen) [inline state-chain/pallets/cf-swapping/src/lib.rs:1058]:
```suggestion
					deposit_amount: amount,
```

The event's deposit_amount should match the amount deposited.

**Review by dandanlen — COMMENTED** [review]
When reviewing this I noticed that the broker's fee is earned even if the swap is rejected. I think we should change this. I'll raise it on Linear.

File: state-chain/pallets/cf-swapping/src/lib.rs (hunk 1 above)
Comment (Janislav) [reply state-chain/pallets/cf-swapping/src/lib.rs:1058]: I don't think this is right. We want to emit the same amount we did before, right? Or was this also wrong before? `amount` is now the raw amount and `net_amount` is `amount - fees`.

**Comment by Janislav** [comment]
Not 100% sure if this is really an issue though. The broker has done his job. Why should he not get his earned fees even if we swallow the swap amount.

File: state-chain/pallets/cf-swapping/src/lib.rs (hunk 1 above)
Comment (dandanlen) [reply state-chain/pallets/cf-swapping/src/lib.rs:1058]: I think it makes more sense to emit the amount deposited in `deposit_amount` (ie. it was wrong before). But maybe this also depends on how it's used in teh swapping app. @acdibble ?

**Comment by dandanlen** [comment]
> Not 100% sure if this is really an issue though. The broker has done his job. Why should he not get his earned fees even if we swallow the swap amount.

Because part of the broker's job is spam prevention.

**Review by dandanlen — APPROVED** [review]

**Pull request merged**